pkg/prometheus/statefulset.go

```diff
@@ -749,14 +750,18 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 			Resources: p.Spec.Thanos.Resources,
 		}
 
-		if p.Spec.Thanos.ObjectStorageConfig != nil {
-			container.Args = append(container.Args, "--objstore.config=$(OBJSTORE_CONFIG)")
-			container.Env = append(container.Env, v1.EnvVar{
-				Name: "OBJSTORE_CONFIG",
-				ValueFrom: &v1.EnvVarSource{
-					SecretKeyRef: p.Spec.Thanos.ObjectStorageConfig,
-				},
-			})
+		if p.Spec.Thanos.ObjectStorageConfig != nil || p.Spec.Thanos.ObjectStorageConfigFile != nil {
+			if p.Spec.Thanos.ObjectStorageConfig != nil {
+				container.Args = append(container.Args, "--objstore.config=$(OBJSTORE_CONFIG)")
+				container.Env = append(container.Env, v1.EnvVar{
+					Name: "OBJSTORE_CONFIG",
+					ValueFrom: &v1.EnvVarSource{
+						SecretKeyRef: p.Spec.Thanos.ObjectStorageConfig,
+					},
+				})
+			} else {
```

Comment:
I would not do the else here and instead keep what we had before and just do something like:

```
if p.Spec.Thanos.ObjectStorageConfigFile != nil {
// append the container args here
}
```

This was we let thanos decide the ordering and if user sets both, both are passed.